File: core/src/com/symbol/ecs/component/map/ClampComponent.java
package com.symbol.ecs.component.map;

import com.badlogic.ashley.core.Component;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.utils.Pool;

public class ClampComponent implements Component, Pool.Poolable {

    public boolean right = false;
    public Rectangle rect = new Rectangle();
    public float acceleration = 0f;
    public float backVelocity = 0f;
    public boolean clamping = true;

    @Override
    public void reset() {
        right = false;
        rect.set(0f, 0f, 0f, 0f);
        acceleration = 0f;
        backVelocity = 0f;
        clamping = true;
    }

}

File: core/src/com/symbol/ecs/component/map/ClampComponent.kt
package com.symbol.ecs.component.map

import com.badlogic.ashley.core.Component
import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.utils.Pool

class ClampComponent : Component, Pool.Poolable {

    var right: Boolean = false
    var rect: Rectangle = Rectangle()
    var acceleration: Float = 0f
    var backVelocity: Float = 0f
    var clamping: Boolean = true

    override fun reset() {
        right = false
        rect.set(0f, 0f, 0f, 0f)
        acceleration = 0f
        backVelocity = 0f
        clamping = true
    }

}